internal/config/gql_provider.go

```diff
@@ -0,0 +1,45 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+)
+
+// GqlProvider is GraphQL provider
+type GqlProvider struct {
+	gqlClient *GqlClient
+}
+
+// NewGqlProvider initializes new GraphQL config source provider
+func NewGqlProvider(gql *GqlClient) *GqlProvider {
+	return &GqlProvider{gqlClient: gql}
+}
+
+// Configs returns list of config files
+func (g *GqlProvider) Configs(ctx context.Context) ([]string, error) {
```

Comment:
Do we need to create file? I know that koanf supports loading config from raw bytes: https://github.com/knadh/koanf#reading-raw-bytes

That would simplify the code 👍

Reply:
I didn't wanted to do so much change, as you know the config struct used in koanf is dedicated to yaml, where I am using json notation for botkubeConfig. That is why I have written that in file and provided as application parameters. So, if I use koanf to unmarshall to our config, I need to add `json` tag to all struct fields.
Instead I have saved the config and reused botkube environment vars to prevent a huge change.

Reply:
What about unmarshalling it by yourself and using Go struct provider https://github.com/knadh/koanf/blob/master/providers/structs/structs.go#L22?

Reply:
Yesterday I paired with Mateusz and restructured it, now it is not generating file, it returns set of config contents instead.